.github/workflows/bazel_test.yml

```diff
@@ -30,6 +35,9 @@ jobs:
             clang-10 \
             libunwind-dev \
             libblocksruntime-dev
+      - name: Use Bazel last_green on schedule
+        if: github.event.schedule == '0 0 * * *'
+        run: echo "USE_BAZEL_VERSION=last_green" >> $GITHUB_ENV
```

Comment:
I was briefly reading about environment variables for Github actions. Have you considered the alternative of an `env` section at the beginning of the job, guarded by an `if` conditional? Any trade-offs?

Reply:
It's possible to set environment variables via `env`, but I thought that didn't work conditionally: The `if` would run the entire job conditionally, not just set the variables. But I don't know GitHub Actions well enough to say with certainty that it's not possible.

Reply:
I found this: https://github.com/actions/runner/issues/409

I'm wondering if this form works at the beginning of the job:

```
jobs:
  unit_tests:
    env:
      USE_BAZEL_VERSION: ${{ github.event.schedule == '' && 'last_green' || '' }}
```

But I'm happy with either form, so let me know what you decide.

Reply:
Great find (and great hack). I verified it works by trying it both this way and with the inverted condition, observing that the Bazel version did indeed change as expected.

Now that we can do this inline in an env, I moved it to the workflow level, which gets rid of all repetitions.

Reply:
> Now that we can do this inline in an env, I moved it to the workflow level, which gets rid of all repetitions.

Excellent, thank you!